app/lib/katello/validators/content_view_environment_validator.rb

```diff
@@ -6,14 +6,27 @@ def validate(record)
         environment_id = record.respond_to?(:lifecycle_environment_id) ? record.lifecycle_environment_id : record.environment_id
 
         if record.content_view_id && environment_id
-          view = ContentView.find(record.content_view_id)
-          env = KTEnvironment.find(environment_id)
-          unless view.in_environment?(env)
+          view = rescue_not_found(record, 'content_view', record.content_view_id) do
+            ContentView.find(record.content_view_id)
+          end
+          env = rescue_not_found(record, 'environment', environment_id) do
+            KTEnvironment.find(environment_id)
```

Comment:
Please change it to `.where(:id => environment_id).first`. It will avoid exception altogether.